specification/botservice/resource-manager/Microsoft.BotService/stable/2021-03-01/botservice.json

```diff
@@ -0,0 +1,2640 @@
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.BotService/botServices/{resourceName}/connections/{connectionName}": {
+      "put": {
+        "tags": [
+          "BotConnection"
+        ],
+        "description": "Register a new Auth Connection for a Bot Service",
+        "operationId": "BotConnection_Create",
+        "x-ms-examples": {
+          "Create Connection Setting": {
+            "$ref": "./examples/PutConnection.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/resourceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/connectionNameParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ConnectionSetting"
+            },
+            "description": "The parameters to provide for creating the Connection Setting."
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "If resource is created successfully or already existed, the service should return 200 (OK).",
+            "schema": {
+              "$ref": "#/definitions/ConnectionSetting"
+            }
+          },
+          "201": {
+            "description": "If resource is created successfully, the service should return 201 (Created). Execution to continue asynchronously.",
+            "schema": {
+              "$ref": "#/definitions/ConnectionSetting"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed",
+            "schema": {
+              "$ref": "#/definitions/Error"
+            }
+          }
+        }
+      },
+      "patch": {
+        "tags": [
+          "BotConnection"
+        ],
+        "description": "Updates a Connection Setting registration for a Bot Service",
+        "operationId": "BotConnection_Update",
+        "x-ms-examples": {
+          "Update Connection Setting": {
+            "$ref": "./examples/UpdateConnection.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/resourceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/connectionNameParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ConnectionSetting"
```

Comment:
Can you patch all of these properties?

Reply:
Again BotConnection -> BotConnections across the baord for these operations, unless that would be a breaking change

Reply:
For connection sub type, all properties are patchable.

Keep as is for back-compat. since according to https://swagger.io/docs/specification/paths-and-operations/ "Some code generators use this value to name the corresponding methods in code."